src/Core/src/Platform/iOS/MauiUIContextMenuInteraction.cs

```diff
@@ -13,13 +13,9 @@ class MauiUIContextMenuInteraction : UIContextMenuInteraction
 	{
 		WeakReference<IElementHandler> _handler;
 
-		// Store a reference to the platform delegate so that it is not garbage collected
-		IUIContextMenuInteractionDelegate? _uiContextMenuInteractionDelegate;
```

Comment:
Looking at the comment, isn't this here for a reason? @Eilon  @PureWeen  do you remember?

Reply:
No, I showed this to the iOS team in a meeting and they say we should delete it. I planned to have @mandel-macaque review when this is green.

Reply:
Well, yeah there's a reason, it says right there 😁 

It's from quite some time ago, but my recollection is that without it (which was my first attempt when doing context menus), the delegate would get GC'ed and then you get no context menus after a short amount of time (because it got GC'ed). Or maybe the menu is there but you can't click or something. I recall it took me quite a lot of debugging to figure that out, and that's what I ended up with.

So, please carefully test context menu scenarios on Mac/iOS before deleting this. My also vague recollection is that it wasn't too difficult to repro: just run a context menu sample (like in the Controls Sample project) and after a short while of playing with context menus, something stops working (either they don't show up at all (?), or they show up but clicking does nothing (?)).

Reply:
@Eilon can we put that in the commit message next time? 😄 

I think the iOS team should review why you need to do this either way.

Reply:
The problem with saving delegates in a field, either:
1. Customers won't know to do this
2. It actually leaks (possible if two objects have strong references to each other)
3. It isn't needed

So, I mainly want to understand which one applies here. We might wait for @rolfbjarne to return from OOF, too.

Reply:
Indeed, perfectly fair points. Unfortunately, I don't remember that far back.

Reply:
It looks like the field is solving an actual problem, because otherwise the delegate is collected.

Apple says it's a weak property: https://developer.apple.com/documentation/uikit/uicontextmenuinteraction/3295925-delegate?language=objc, so unless managed code keeps a reference to it, it'll be collected.

It doesn't look like there should be a leak here though, because the delegate instance doesn't have any references to anything else.

However, there's a race condition, the managed FlyoutUIContextMenuInteractionDelegate instance is created in the call to the base constructor, then fetched in the next line. If the GC runs between those two lines, the delegate can still be collected.

Ideally it would be possible to fix this by making `MauiUIContextMenuInteraction` implement `IUIContextMenuInteractionDelegate` and the corresponding interface method:

```cs
class MauiUIContextMenuInteraction : UIContextMenuInteraction, IUIContextMenuInteractionDelegate
{
	public MauiUIContextMenuInteraction (IElementHandler handler)
		: base (this)
	{ /* ... */ }
	
	UIContextMenuConfiguration? IUIContextMenuInteractionDelegate.GetConfigurationForMenu(UIContextMenuInteraction interaction, CGPoint location)
	{ /* ... */ }
}
```

but that runs into:

> error CS0027: Keyword 'this' is not available in the current context

But I believe his should fix the race condition and keep the delegate alive:

```cs
class MauiUIContextMenuInteraction : UIContextMenuInteraction, IUIContextMenuInteractionDelegate
{
	IUIContextMenuInteractionDelegate? _uiContextMenuInteractionDelegate;
	public MauiUIContextMenuInteraction (IElementHandler handler)
		: base (CreateDelegate (out var del)
	{
		_uiContextMenuInteractionDelegate = del;
	}

	static IUIContextMenuInteractionDelegate CreateDelegate (out IUIContextMenuInteractionDelegate del)
	{
		return del = new FlyoutUIContextMenuInteractionDelegate();
	}
}
```

Reply:
@rolfbjarne I'll update the code for this one, but that brings up two questions:

1. Is there a `<summary>` comment or something that would tell developers this delegate is weak? And then maybe link to a docs page that explains more?

2. Could the `UIContextMenuInteraction` automatically save the instance in a field for you somehow? Would be a change to binding generation, I guess. And then any manual bindings are a problem...

Reply:
This class is a bit of a weird case (passing the delegate as a parameter to the constructor is not common, it's usually a property setter), so I've created this fix: https://github.com/xamarin/xamarin-macios/pull/19386